internal/ctxval/context_values.go

```diff
@@ -20,26 +20,35 @@ const (
 	requestIdCtxKey      commonKeyId = iota
 	accountIdCtxKey      commonKeyId = iota
 	unleashContextCtxKey commonKeyId = iota
+	identityCtxKey       commonKeyId = iota
 )
 
 var MissingAccountInContextError = errors.New("operation requires account_id in context")
 
-// Identity returns identity header struct or nil when not set.
-func Identity(ctx context.Context) identity.XRHID {
-	return identity.Get(ctx)
-}
-
 // Logger returns the main logger with context fields or the standard global logger
-// when the main logger was not set.
+// when the main logger was not set. Never returns nil.
 func Logger(ctx context.Context) *zerolog.Logger {
 	if ctx == nil || ctx.Value(loggerCtxKey) == nil {
 		return &log.Logger
 	}
 	return ctx.Value(loggerCtxKey).(*zerolog.Logger)
 }
 
+func WithLogger(ctx context.Context, logger *zerolog.Logger) context.Context {
+	return context.WithValue(ctx, loggerCtxKey, logger)
+}
+
+// Identity returns identity header struct or nil when not set.
+func Identity(ctx context.Context) identity.XRHID {
+	value := ctx.Value(identityCtxKey)
+	if value == nil {
+		return identity.XRHID{}
+	}
+	return value.(identity.XRHID)
+}
+
 func WithIdentity(ctx context.Context, id identity.XRHID) context.Context {
-	return context.WithValue(ctx, identity.Key, id)
+	return context.WithValue(ctx, identityCtxKey, id)
```

Comment:
I liked to use the shared key, so we could pass our context to other shared code, but if you insist :shrug: :)

Reply:
Context should not be used to pass arguments through API boundaries, I have never seen that either. It is a small change for consistency, if we ever hit this issue we can copy identity with the needed key.

Reply:
I remember that it is an issue to use a different key than the shared one because sources look for the identity in the shared key. 
Have you checked that the workflow stayed intact? 
When I checked your PR I got `11:48AM ERR service returned unauthorized ` while on the main branch the launch instance job has succeeded.

Reply:
Yeah it looks like some bug. But the key should not matter, it is our own context for our own app only. I bet it is something else, let me take a look.